core/src/main/java/org/apache/calcite/rex/RexExecutable.java

```diff
@@ -87,11 +88,14 @@ public void reduce(RexBuilder rexBuilder, List<RexNode> constExps,
         values = new Object[constExps.size()];
       } else {
         assert values.length == constExps.size();
+        final List<RexNode> successfullyReduced = new ArrayList<>();
         final List<@Nullable Object> valueList = Arrays.asList(values);
         for (Pair<RexNode, @Nullable Object> value : Pair.zip(constExps, valueList)) {
-          reducedValues.add(
+          successfullyReduced.add(
               rexBuilder.makeLiteral(value.right, value.left.getType(), true));
         }
+        assert successfullyReduced.size() == constExps.size();
```

Comment:
`values.length == constExps.size();` and `successfullyReduced.size() == values.length`.

Should we need `assert successfullyReduced.size() == constExps.size();`?

Reply:
@JiajunBernoulli  sorry I don't quite understand your comment?, wdym?

Reply:
Maybe `assert successfullyReduced.size() == constExps.size()` is redundant.

Reply:
Certainly I can't imagine a scenario where that equality won't hold (if the `Pair#zip` behaves as it should), but the assertion doesn't hurt (and assertions can be disabled anyway), so I'm not against having it.